### PR TITLE
chore: Clean up and convert field-bitmap to Typescript

### PR DESCRIPTION
**Pull request by gonfunko**
## The basics

- [x] I [validated my changes](https://developers.google.com/blockly/guides/contribute/samples#making_and_verifying_a_change)

### Resolves
Part of #1647

### Proposed Changes
This PR converts field-bitmap to Typescript. It also includes some light refactoring and cleanup of the code.

### CODE REVIEW

File: plugins/field-bitmap/src/field-bitmap.ts [hunk 1]
```diff
@@ -0,0 +1,595 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Bitmap input field.
```
Comment (maribethb) [inline plugins/field-bitmap/src/field-bitmap.ts]: This fileoverview isn't adding anything and we no longer include author annotations so I would remove this comment

File: plugins/field-bitmap/src/field-bitmap.ts [hunk 2]
```diff
@@ -0,0 +1,595 @@
...
+ * @author gregoryc@outlook.com (Greg Cannon)
+ */
+
+import Blockly from 'blockly/core';
+
+export const DEFAULT_HEIGHT = 5;
```
Comment (maribethb) [inline plugins/field-bitmap/src/field-bitmap.ts:9]: note on the lint errors: rachel has an update to the eslint config that allows constant case. it's bundled in another pr. I will poke her about submitting that when she is feeling better

File: plugins/field-bitmap/src/field-bitmap.ts [hunk 3]
```diff
@@ -0,0 +1,595 @@
...
+      boundingBox.right,
+    );
+  }
+
+  /**
+   * Creates the bitmap editor and add event listeners.
+   *
+   * @returns The newly created dropdown menu.
+   */
+  private dropdownCreate_() {
```
Comment (maribethb) [inline plugins/field-bitmap/src/field-bitmap.ts]: do you also want to rename this one to remove the underscore?

**Review by maribethb — APPROVED** [review]

File: plugins/field-bitmap/src/field-bitmap.ts (hunk 1 above)
Comment (gonfunko) [reply plugins/field-bitmap/src/field-bitmap.ts]: Done!

File: plugins/field-bitmap/src/field-bitmap.ts (hunk 3 above)
Comment (gonfunko) [reply plugins/field-bitmap/src/field-bitmap.ts]: Good catch, thanks! Renamed accordingly.

File: plugins/field-bitmap/src/field-bitmap.ts (hunk 2 above)
Comment (gonfunko) [reply plugins/field-bitmap/src/field-bitmap.ts:9]: Ah nice, and sounds good!

**Pull request merged**